### PR TITLE
Add section to editors guide on use of PREFIX and BASE

### LINKED ISSUE #88: add something to the wiki on conventions. [1]
Opened by RRSAgent via IRC channel #rdf-star on irc.w3.org

Due: 2023-09-19 (Tuesday 19 September)

### PR DESCRIPTION
**Pull request by gkellogg**
Fixes #88

### CODE REVIEW

**Review by Tpt — APPROVED** [review]

**Review by hartig — APPROVED** [review]

**Review by domel — APPROVED** [review]

File: docs/editors-guide.md [hunk 1]
```diff
@@ -102,6 +102,34 @@ Consider using `data-transform="updateExample"` with `<!-- .. -->` comments to a
...
+
+<#spiderman>
+  rel:enemyOf <#green-goblin> ;
+  a foaf:Person ;
+  foaf:name "Spiderman", "Человек-паук"@ru .
+```
+
+(There was some preference for using the lower-case `prefix` and `base` instead, which is allowed by the grammar, as such tokens are case-insensitive, but consistency is also important).
+
+Exceptions to this rule is where the grammar is discussed explicitly, and examples using `@prefix` and `@base` are appropriate. Also, possibly `prefix` and `base` to clarify the case-insensitive nature of such tokens.
```
Comment (TallTed) [inline docs/editors-guide.md:131]:
```suggestion
Exceptions to this rule are where the grammar is discussed explicitly, and examples using `@prefix` and `@base` are appropriate. Also, possibly `prefix` and `base` to clarify the case-insensitive nature of such tokens.
```

File: docs/editors-guide.md [hunk 2]
```diff
@@ -102,6 +102,34 @@ Consider using `data-transform="updateExample"` with `<!-- .. -->` comments to a
   </pre>
 ```
 
+### PREFIX/BASE use in examples
```
Comment (TallTed) [inline docs/editors-guide.md:105]:
```suggestion
### Using `PREFIX` and `BASE` in examples
```

File: docs/editors-guide.md [hunk 3]
```diff
@@ -102,6 +102,34 @@ Consider using `data-transform="updateExample"` with `<!-- .. -->` comments to a
...
+
+Turtle and TriG examples should prefer the use of `PREFIX` and `BASE` to `@prefix` and `@base`. This creates a uniform style between Turtle/TriG and SPARQL and facilitates copy/paste.
```
Comment (TallTed) [inline docs/editors-guide.md:107]: Turtle and SPARQL already support both forms (i.e., anyone can use any of `PREFIX`, `BASE`, `@prefix`, or `@base`); I think TriG does as well.

I was not aware that all of these forms were case insensitive (i.e., that `PREFIX`, `@PREFIX`, `@prefix`, `prefix`, and I presume any oddmix, like `PrEfIx`, are also valid). This should perhaps be made more explicit in the spec texts.

**Review by TallTed — CHANGES_REQUESTED** [review]

File: docs/editors-guide.md (hunk 3 above)
Comment (gkellogg) [reply docs/editors-guide.md:107]: SPARQL does not support `@prefix`. TriG does, of course, and any examples should be updated, but I don't think it necessary to highlight the use of `@prefix` or indeed `prefix` in the TriG grammar section given that it is explicitly derivative of Turtle. If it proves confusing, we can re-examine this.

`@prefix` is case sensitive, but `PREFIX` is not. Odd mixes of the `PREFIX` form are also acceptable, and I think we have tests for this. I think it's sufficient to say that this form is case-insensitive with an example using `prefix` and `base` to demonstrate.

**Pull request merged**